cub/cub/device/dispatch/kernels/transform.cuh

```diff
@@ -31,7 +32,38 @@
 #endif
 
 CUB_NAMESPACE_BEGIN
+namespace detail::transform
+{
+struct thread_block
+{
+  _CCCL_DEVICE int size() const
+  {
+    return blockDim.x;
+  }
+
+  _CCCL_DEVICE int thread_rank() const
+  {
+    return threadIdx.x;
+  }
 
+  _CCCL_DEVICE void sync() const
+  {
+    __syncthreads();
+  }
+};
+} // namespace detail::transform
+CUB_NAMESPACE_END
+
+_CG_BEGIN_NAMESPACE
+namespace details
+{
+template <>
+struct _async_copy_group_supported<CUB_NS_QUALIFIER::detail::transform::thread_block> : _CG_STL_NAMESPACE::true_type
```

Comment:
Question: Why are we using that funky `true_type`

Reply:
Because I am hacking myself through CG.